### DISCUSSION TITLE
Is there a way to read `task` output from the terminal?

### DISCUSSION THREAD

**Discussion by prashantvc** [microsoft/vscode-discussions#682] [Extension Development QnA] [answered]
I am trying to read one of my extensions task output from the terminal, and I am unable to find any API support for it.
Is there a way to read the output log from the terminal? The only option I have right now is to write the output to a file and read it, or start a new process

**Comment by vaclavHala** [comment, marked as answer]
Hi, I was facing the same problem and came up with this. It is a bit of a hack but if you don't have to care about multiple tasks executing in parallel it should work fine

``` ts
function awaitTaskResult(taskName: string, timeoutMillis: number): Promise<string> {
    let resolve: (output: string) => void;
    let reject: (err: any) => void;
    const taskResult = new Promise<string>((_resolve, _reject) => {
        resolve = _resolve;
        reject = _reject;
    });
    const endListener = vscode.tasks.onDidEndTask(taskEnded => {
        if (taskEnded.execution.task.name === taskName) {
            vscode.commands.executeCommand("workbench.action.terminal.copyLastCommandOutput")
                .then(
                    ok => vscode.env.clipboard.readText(),
                    err => reject(err))
                .then(output => resolve(output))
        }
    });
    taskResult.finally(() => {
        endListener.dispose()
    });
    setTimeout(() => reject(new Error("Task did not finish in time")), timeoutMillis);
    return taskResult
}
```

if anyone knows about better way I'd be glad to hear it too

**Reply by prashantvc** [reply]
Thanks, this is neat. I figured it's just easier to spawn a task as a process and read the output.
@isidorn I don't know if it's already proposed, it will be a good addition to task APIs

**Reply by isidorn** [reply]
Thanks @prashantvc for the ping.
fyi @meganrogge in case there is a neater solution for this.

**Reply by meganrogge** [reply]
There isn't currently a way to do this, so we'd have to add API
